src/Tuura/Concept/Circuit/Basic.hs

```diff
@@ -35,17 +36,19 @@ data Concept s e a = Concept
                    {
                        initial   :: a -> InitialValue,
                        arcs      :: [([e], e)],
-                       interface :: a -> Interface
+                       interface :: a -> Interface,
+                       invariant :: [[e]]
                    }
```

Comment:
That's fair enough. I was planning on doing something similar. I will include this in the next iteration.